### PR TITLE
Зачатки тестов

### PR DESCRIPTION
**Pull request by dontuse**
Заготовка на обсуждение.
ссылки по теме 
https://facebook.github.io/jest/
https://github.com/airbnb/enzyme
https://hackernoon.com/testing-react-components-with-jest-and-enzyme-41d592c174f
http://redux.js.org/docs/recipes/WritingTests.html
https://youtu.be/59Ndb3YkLKA

### CODE REVIEW

File: test/setup.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// ToDo: Запилить сетап
+import app from '../src/_app.mock';
+
+global.app = app;
```
Comment (apress-hound) [inline test/setup.js]: 'global' is not defined.

File: src/ImageEditor/__tests__/saga.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {updateImages} from '../sagas';
+import {call, put, take} from 'redux-saga/effects';
+import nock from 'nock';
```
Comment (apress-hound) [inline src/ImageEditor/__tests__/saga.js]: 'nock' is defined but never used.

File: src/ImageEditor/__tests__/saga.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {updateImages} from '../sagas';
+import {call, put, take} from 'redux-saga/effects';
```
Comment (apress-hound) [inline src/ImageEditor/__tests__/saga.js]: 'call' is defined but never used.<br>'put' is defined but never used.<br>'take' is defined but never used.

File: src/ImageEditor/__tests__/saga.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import nock from 'nock';
+
+describe('', () => {
+  const generator = updateImages({payload: {images: [], files: [{src: '', id: 0}]}});
+
+  it('просто бесполезный тест 1', () => {
+    const testValue = generator.next().value;
```
Comment (apress-hound) [inline src/ImageEditor/__tests__/saga.js]: 'testValue' is defined but never used.

File: src/Button/__tests__/Button.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import {shallow, mount, render} from 'enzyme';
+import Button from '../Button';
+
+describe('Button', () => {
+  it('should render a Button', () => {
+    const wrapper = shallow(
+      <Button>Some text</Button>
```
Comment (apress-hound) [inline src/Button/__tests__/Button.js]: Expected an identifier and instead saw '<'.<br>Expected ')' and instead saw 'Button'.<br>Missing semicolon.<br>Unclosed regular expression.<br>Unrecoverable syntax error. (27% scanned).

**Comment by Napolskih** [comment]
сборки удалил. а когда и кто собирать будет?

**Comment by dontuse** [comment]
это просто на обсудить в понедельник, сборки не нужны. Соберу если будем писать тесты на jest и enzyme.

**Comment by dontuse** [comment]
:nomerge:

File: coverage/lcov-report/sorter.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+            return;
+        }
+        cols = loadColumns();
+        loadData(cols);
+        addSortIndicators();
+        enableUI();
+    };
+})();
+
+window.addEventListener('load', addSorting);
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: coverage/lcov-report/sorter.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            };
+        for (i =0 ; i < cols.length; i += 1) {
+            if (cols[i].sortable) {
+                // add the click event handler on the th so users
+                // dont have to click on those tiny arrows
+                el = getNthColumn(i).querySelector('.sorter').parentElement;
+                if (el.addEventListener) {
+                    el.addEventListener('click', ithSorter(i));
+                } else {
+                    el.attachEvent('onclick', ithSorter(i));
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: coverage/lcov-report/sorter.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+                    addSortIndicators();
+                };
+            };
+        for (i =0 ; i < cols.length; i += 1) {
+            if (cols[i].sortable) {
+                // add the click event handler on the th so users
+                // dont have to click on those tiny arrows
+                el = getNthColumn(i).querySelector('.sorter').parentElement;
+                if (el.addEventListener) {
+                    el.addEventListener('click', ithSorter(i));
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: coverage/lcov-report/sorter.js [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+                    currentSort.index = i;
+                    currentSort.desc = desc;
+                    addSortIndicators();
+                };
+            };
+        for (i =0 ; i < cols.length; i += 1) {
+            if (cols[i].sortable) {
+                // add the click event handler on the th so users
+                // dont have to click on those tiny arrows
+                el = getNthColumn(i).querySelector('.sorter').parentElement;
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: coverage/lcov-report/sorter.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+    function removeSortIndicators() {
+        var col = getNthColumn(currentSort.index),
+            cls = col.className;
+
+        cls = cls.replace(/ sorted$/, '').replace(/ sorted-desc$/, '');
+        col.className = cls;
+    }
+    // adds sort indicators for current column being sorted
+    function addSortIndicators() {
+        getNthColumn(currentSort.index).className += currentSort.desc ? ' sorted-desc' : ' sorted';
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: coverage/lcov-report/sorter.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+        for (i = 0; i < rows.length; i += 1) {
+            tableBody.appendChild(rows[i]);
+        }
+    }
+    // removes sort indicators for current column being sorted
+    function removeSortIndicators() {
+        var col = getNthColumn(currentSort.index),
+            cls = col.className;
+
+        cls = cls.replace(/ sorted$/, '').replace(/ sorted-desc$/, '');
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: coverage/lcov-report/sorter.js [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+    function sortByIndex(index, desc) {
+        var key = cols[index].key,
+            sorter = function (a, b) {
+                a = a.data[key];
+                b = b.data[key];
+                return a < b ? -1 : a > b ? 1 : 0;
+            },
+            finalSorter = sorter,
+            tableBody = document.querySelector('.coverage-summary tbody'),
+            rowNodes = tableBody.querySelectorAll('tr'),
```
Comment (apress-hound) [inline coverage/lcov-report/sorter.js]: Mixed double and single quotes.

File: test/setup.js (hunk 1 above)
Comment (apress-hound) [inline test/setup.js]: 'global' is not defined.

File: src/ImageEditor/__tests__/saga.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
+import {updateImages} from '../sagas';
+import {call, put, take, select} from 'redux-saga/effects';
```
Comment (apress-hound) [inline src/ImageEditor/__tests__/saga.js]: 'call' is defined but never used.<br>'take' is defined but never used.<br>'select' is defined but never used.

File: test/setup.js (hunk 1 above)
Comment (apress-hound) [inline test/setup.js]: 'global' is not defined.

File: src/ImageEditor/__tests__/saga.js (hunk 13 above)
Comment (apress-hound) [inline src/ImageEditor/__tests__/saga.js]: 'call' is defined but never used.<br>'take' is defined but never used.<br>'select' is defined but never used.

File: src/Button/__tests__/Button.js (hunk 5 above)
Comment (apress-hound) [inline src/Button/__tests__/Button.js]: Expected an identifier and instead saw '<'.<br>Expected ')' and instead saw 'Button'.<br>Missing semicolon.<br>Unclosed regular expression.<br>Unrecoverable syntax error. (27% scanned).

File: dist/assets-help.bundle.js [hunk 14]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 		var me = installedModules[moduleId];
+/******/ 		if(!me) return __webpack_require__;
+/******/ 		var fn = function(request) {
+/******/ 			if(me.hot.active) {
+/******/ 				if(installedModules[request]) {
+/******/ 					if(installedModules[request].parents.indexOf(moduleId) < 0)
+/******/ 						installedModules[request].parents.push(moduleId);
+/******/ 					if(me.children.indexOf(request) < 0)
+/******/ 						me.children.push(request);
+/******/ 				} else hotCurrentParents = [moduleId];
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: Expected '{' and instead saw 'hotCurrentParents'.<br>Too many errors. (0% scanned).

File: dist/assets-help.bundle.js [hunk 15]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 	function hotCreateRequire(moduleId) { // eslint-disable-line no-unused-vars
+/******/ 		var me = installedModules[moduleId];
+/******/ 		if(!me) return __webpack_require__;
+/******/ 		var fn = function(request) {
+/******/ 			if(me.hot.active) {
+/******/ 				if(installedModules[request]) {
+/******/ 					if(installedModules[request].parents.indexOf(moduleId) < 0)
+/******/ 						installedModules[request].parents.push(moduleId);
+/******/ 					if(me.children.indexOf(request) < 0)
+/******/ 						me.children.push(request);
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: Expected '{' and instead saw 'me'.

File: dist/assets-help.bundle.js [hunk 16]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 	var hotCurrentParents = []; // eslint-disable-line no-unused-vars
+/******/ 	
+/******/ 	function hotCreateRequire(moduleId) { // eslint-disable-line no-unused-vars
+/******/ 		var me = installedModules[moduleId];
+/******/ 		if(!me) return __webpack_require__;
+/******/ 		var fn = function(request) {
+/******/ 			if(me.hot.active) {
+/******/ 				if(installedModules[request]) {
+/******/ 					if(installedModules[request].parents.indexOf(moduleId) < 0)
+/******/ 						installedModules[request].parents.push(moduleId);
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: Expected '{' and instead saw 'installedModules'.

File: dist/assets-help.bundle.js [hunk 17]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 	}
+/******/ 	
+/******/ 	var hotApplyOnUpdate = true;
+/******/ 	var hotCurrentHash = "4768a145682bce55229b"; // eslint-disable-line no-unused-vars
+/******/ 	var hotCurrentModuleData = {};
+/******/ 	var hotCurrentParents = []; // eslint-disable-line no-unused-vars
+/******/ 	
+/******/ 	function hotCreateRequire(moduleId) { // eslint-disable-line no-unused-vars
+/******/ 		var me = installedModules[moduleId];
+/******/ 		if(!me) return __webpack_require__;
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: Expected '{' and instead saw 'return'.<br>Identifier '__webpack_require__' is not in camel case.

File: dist/assets-help.bundle.js [hunk 18]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 		Object.defineProperty({}, "x", {
+/******/ 			get: function() {}
+/******/ 		});
+/******/ 		canDefineProperty = true;
+/******/ 	} catch(x) {
+/******/ 		// IE will fail on defineProperty
+/******/ 	}
+/******/ 	
+/******/ 	var hotApplyOnUpdate = true;
+/******/ 	var hotCurrentHash = "4768a145682bce55229b"; // eslint-disable-line no-unused-vars
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: 'hotCurrentHash' was used before it was defined.<br>Mixed double and single quotes.

File: dist/assets-help.bundle.js [hunk 19]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 			}
+/******/ 		};
+/******/ 	}
+
+/******/ 	
+/******/ 	
+/******/ 	// Copied from https://github.com/facebook/react/blob/bef45b0/src/shared/utils/canDefineProperty.js
+/******/ 	var canDefineProperty = false;
+/******/ 	try {
+/******/ 		Object.defineProperty({}, "x", {
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: Mixed double and single quotes.

File: dist/assets-help.bundle.js [hunk 20]
```diff
@@ -13,537 +13,537 @@
...
+/******/ 			if(request.readyState !== 4) return;
+/******/ 			if(request.status === 0) {
+/******/ 				// timeout
+/******/ 				callback(new Error("Manifest request to " + requestPath + " timed out."));
+/******/ 			} else if(request.status === 404) {
+/******/ 				// no update available
+/******/ 				callback();
+/******/ 			} else if(request.status !== 200 && request.status !== 304) {
+/******/ 				// other failure
+/******/ 				callback(new Error("Manifest request to " + requestPath + " failed."));
```
Comment (apress-hound) [inline dist/assets-help.bundle.js]: Mixed double and single quotes.

**Pull request merged**